test/SparqlAntlrParserTest.cpp

```diff
@@ -44,11 +44,46 @@ auto parsePnameNs = parse<&Parser::pnameNs>;
 auto parsePrefixDecl = parse<&Parser::prefixDecl>;
 auto parsePrefixedName = parse<&Parser::prefixedName>;
 auto parsePropertyListPathNotEmpty = parse<&Parser::propertyListPathNotEmpty>;
+auto parseRdfLiteral = parse<&Parser::rdfLiteral>;
 auto parseSelectClause = parse<&Parser::selectClause>;
 auto parseSolutionModifier = parse<&Parser::solutionModifier>;
 auto parseTriplesSameSubjectPath = parse<&Parser::triplesSameSubjectPath>;
 auto parseVerbPathOrSimple = parse<&Parser::verbPathOrSimple>;
 
+namespace {
+template <auto parseFunction>
+auto makeExpectCompleteParser(SparqlQleverVisitor::PrefixMap&& prefixMap = {}) {
+  // Capture `prefixMap` by value. Since the inner code will only be called
+  // later `prefixMap` would be NULL if it were captured by reference.
+  return [prefixMap](const string& input, const auto& matcher) {
+    return expectCompleteParse(parseFunction(input, prefixMap), matcher);
+  };
+}
+
+// TODO: type value can probably removed with some decltype, declvals.
+// TODO: I would like to parametrize the testing::Eq. This seems to not bee
+//  possible because matchers take parameters (so not passable as parameter) and
+//  themselves take a different number of template arguments.
```

Comment:
Ill have a try myself at the first one.

Do you have an Idea how one could go about doing the second one. The goal with this one is to be able to specify a default matcher. It would be nice to say that everything should be matched with `testing::Eq` or `IsVariableSelect`.

Reply:
Added a default type for `Value`.

Since the 2 point does only really make sense when many matchers are lambda I would look at this point in #767.